pinot-filesystem/src/main/java/com/linkedin/pinot/storage/PinotFS.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.storage;
+
+import java.io.IOException;
+import java.net.URI;
+import org.apache.commons.configuration.Configuration;
+
+/**
+ * The PinotFS is intended to be a thin wrapper on top of different filesystems. This interface is intended for internal
+ * Pinot use only. This class will be implemented for each pluggable storage type.
+ */
+public abstract class PinotFS {
+  /**
+   * Initializes the configurations specific to that filesystem. For instance, any security related parameters can be
+   * initialized here and will not be logged.
+   */
+  public abstract void init(Configuration config);
+
+  /**
+   * Deletes the file at the location provided. If the segmentUri is a directory, it will delete the entire directory.
+   * @param segmentUri URI of the segment
+   * @return true if delete is successful else false
+   * @throws IOException IO failure
+   */
+  public abstract boolean delete(URI segmentUri) throws IOException;
+
+  /**
+   * Moves the file from the src to dst. Does not keep the original file. If the dst has parent directories
+   * that haven't been created, this method will create all the necessary parent directories. If dst already exists,
+   * it will overwrite. Will work either for moving a directory or a file.
+   * @param srcUri URI of the original file
+   * @param dstUri URI of the final file location
+   * @return true if move is successful
+   * @throws IOException on failure
+   */
+  public abstract boolean move(URI srcUri, URI dstUri) throws IOException;
```

Comment:
How would move/copy work if the URIs refer to different filesystems?

Reply:
move assumes the same fs, only copyToLocal and copyFromLocal will be allowed for differnet filesystems

Reply:
Will an error be thrown for different schemes in the URI or different destinations?